### DISCUSSION TITLE
Refer to `data-files` from parent directory

### DISCUSSION THREAD

**Discussion by shivaraj-bh** [srid/haskell-flake#352] [Q&A]
For a project with the following structure:
```sh
 .
├──  cabal.project
├──  data.txt
├──  pkg1
│   └──  pkg1.cabal
└──  pkg2
    └──  pkg2.cabal
```

IIUC sub-packages cannot access files in parent directory by design, what are the alternatives that doesn’t involve copying the `data.txt` to every sub-package?

**Comment by roberth** [comment]
Such a project structure is not compatible with hackage either, if that is a goal for you.
Is `cabal-install` happy about something like `data-files: ../data.txt`?
If so, I wouldn't be surprised if `cabal2nix` needs changes to support that.
I haven't tried this.

**Reply by shivaraj-bh** [reply]
`cabal-install` doesn’t complain about it

**Comment by srid** [comment]
> doesn’t involve copying the data.txt to every sub-package?

You don't need to duplicate it in git repo, but you can copy it as part of the source derivations, by overriding:

https://github.com/srid/haskell-flake/blob/66dcee2bfb070bbba3df33cf736a4c0f3ae52b0d/nix/modules/project/packages/package.nix#L16-L24
